scripts/createExercise.js

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env node
+
+//
+// This script takes an exercise folder as a template and creates
+// a new exercise folder from it renaming all files (+file content) accordingly.
+//
+// Example Usage:
+// ./scripts/createcreateExercise.js src/exercises/01-Query-Renderer/ src/exercises/02-New-Exercise
+//
+
+const { execSync } = require("child_process")
+const fs = require("fs")
+const { exit } = require("node:process")
+const { join } = require("path")
+const path = require("path")
+
+const getFileBase = folder => path.parse(folder).base
+
+const renameFileContent = (file, search, replace) => {
+  const data = fs.readFileSync(file, "utf8")
+
+  const result = data.replace(new RegExp(search, "g"), replace)
+
+  fs.writeFileSync(file, result, "utf8")
+}
+
+const rename = (directory, oldNumber, newNumber) => {
+  const files = fs.readdirSync(directory)
+
+  files.forEach(file => {
+    const filePath = join(directory, file)
+    const newFilePath = join(directory, file.replace(oldNumber, newNumber))
+
+    if (fs.lstatSync(filePath).isDirectory()) {
+      // don't rename inside of docs folder
+      if (filePath.endsWith("docs")) return
+      // rename recursively
+      rename(filePath, oldNumber, newNumber)
```

Comment:
It is probably more useful to copy subfolders but leave them empty. 

* The `docs` folder will contain images for the previous exercise
* The `beginning` folder will contain commented-out snapshots of code from the beginning of the previous exercise
* The `completed` folder will contain commented-out snapshots of code from the end of the previous exercise
* The __generated__ folder would get re-generated as soon as we ran relay the first time

The only of those folders that are valuable for a new exercise is the "completed", which would become the "beginning" in the new exercise -- though it might not even be worth preserving that since we could copy from the root to the "beginning" in the new exercise.

Reply:
The other thing that gets copied is the README.md, which will contain instructions that are out of date. I'm cool with just assuming we'll go in there and clear it out manually, but if you feel like emptying that file would make sense to make that part of this PR I'm good with it too.

Reply:
I changed the script and now it:
- creates empty folders `docs`, `beginning` and `completed`
- creates all README.md files
- copies all files from the root and renames them accordingly

The root `README.md` will get copied with content still, but clearing it out manually shouldn't be too bad :)